proto/v1/api/android.proto

```diff
@@ -0,0 +1,391 @@
+  //
+  // Note also that all returned structures are expected to be localized to English by default with
+  // optional additional language localizations available (as limited by the languages proto
+  // version which denotes which languages are currently supported by the API).
+  repeated VersionedStructureDetails current_downloads_context = 4;
+}
+
+// TODO: add support for upcoming topics, for the topic graph ordering, and for topics that can't be downloaded/updated (to provide notices to the user).
+
+// The expected response from the server when provided with a TopicListRequest.
+message TopicListResponse {
+  option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_LIST_REQUEST_RESPONSE_PROTO_VERSION;
+
+  // The common proto version defining this request/response pair. This is expected to match the
+  // version provided in the compatibility context included in the topic list request.
+  org.oppia.proto.v1.versions.TopicListRequestResponseProtoVersion proto_version = 1;
+
+  // A list of all topics that are compatibly available to the client whose context corresponds to
+  // this response. Note that topics marked as available for downloading or updating will always be
+  // compatible with the local client, and topics that are marked as up-to-date are guaranteed to
+  // be at the latest compatible version for the local client. Clients are also expected to lean on
+  // existing local resources for topics that aren't yet downloaded.
+  repeated AvailableTopic available_topics = 2;
+
+  // A topic that's available to & compatible with the local client.
+  message AvailableTopic {
+    option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_LIST_REQUEST_RESPONSE_PROTO_VERSION;
+
+    // The nature of the topic's availability.
+    oneof availability_type {
+      // Indicates a topic that's not yet downloaded by the local client, but is available to be
+      // downloaded.
+      DownloadableTopic downloadable_topic = 1;
+
+      // Indicates a topic that's downloaded by the local client, but has a compatible update.
+      UpdatableTopic updatable_topic = 2;
+
+      // Indicates a topic that's both downloaded by the local client & full up-to-date. This mainly
+      // serves as a lightweight indicator to ensure that the server & client are properly in-sync
+      // with the state of the local client.
+      UpToDateTopic up_to_date_topic = 3;
+    }
+  }
+
+  // Represents a topic that's available to download and is not yet downloaded by the local client.
+  message DownloadableTopic {
+    option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_LIST_REQUEST_RESPONSE_PROTO_VERSION;
+
+    // The ID of the topic that's available to newly download. This is expected to be used to
+    // identify the local copy of the topic's summary in the event that the topic_summary field is
+    // empty. Note that content_version is not also included since topic_summary is only missing if
```

Comment:
I don't think I understand the details of the connection implied here between content_version and topic_summary. I think the question on my mind is "why do we include it in UpdatableTopic and not here", the explanation could perhaps focus more on that.